core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -6587,6 +6587,27 @@ public void testCaseExprForEEE() {
             + "WHERE TO_VARCHAR(\"HIREDATE\", 'DY')";
     assertThat(toSql(root, DatabaseProduct.SNOWFLAKE.getDialect()), isLinux(expectedSF));
   }
+
+  @Test
+  public void testCaseExprForLnSynapse() {
+    final String query = "SELECT LN(\"product_id\") as dd from \"product\"";
+    final String expectedSynapse = "SELECT LOG([product_id]) AS [DD]"
+            + "\nFROM [foodmart].[product]";
+    sql(query)
+            .withMssql()
+            .ok(expectedSynapse);
+  }
+
+  @Test
+  public void testCaseExprForLog10Synapse() {
+    final String query = "SELECT LOG10(\"product_id\") as dd from \"product\"";
+    final String expectedSynapse = "SELECT LOG10([product_id]) AS [DD]"
+            + "\nFROM [foodmart].[product]";
+    sql(query)
+            .withMssql()
+            .ok(expectedSynapse);
+  }
+
```

Comment:
Add a test case for 'Cast' as well

Reply:
Not Possible. There is no such table in calcite which has varchar type column with numeric values.